myhpi/core/middleware.py

```diff
@@ -20,10 +20,9 @@ def __call__(self, request):
         return response
 
     def process_request(self, request):
-        if request.user.is_anonymous:
-            address = ip_address(request.META.get("REMOTE_ADDR"))
-            for ip_range, group_name in self.ip_ranges.items():
-                if address in ip_range:
-                    # user is in this IP range
-                    request.user._ip_range_group_name = group_name
-                    break
+        address = ip_address(request.META.get("REMOTE_ADDR"))
+        for ip_range, group_name in self.ip_ranges.items():
+            if address in ip_range:
+                # user is in this IP range
+                request.user._ip_range_group_name = group_name
+                break
```

Comment:
Out of curiosity: What is the behaviour of the new code? Do I need to be in the University network even when I am logged in?

Reply:
Depending on which IP the user has, they are assigned a certain role in addition to the roles that they might already have. If the user is somewhere in the UP network, they might receive a role such as "UP" or any the like. That will affect which of the pages will be visible to the user.
Previously the user would only receive that additional role when being anonymous, which is an issue, because whenever you signed in to your account you were not anonymous anymore, leading to the loss of that role - even if you're still in the same network. And it somehow would be weird if you can view less pages whenever you sign in (kinda counter-intuitive).

Reply:
Ohh, I see now: Looks like we can define one role per specific IP range that is then added to the user?
A little comment (or maybe only an extraction method like `assign_group_for_ip_range`) would have made that clearer to me